src/shims/time.rs

```diff
@@ -4,13 +4,19 @@ use crate::stacked_borrows::Tag;
 use crate::*;
 use helpers::int_to_immty_checked;
 
+// Extracts the number of seconds and nanoseconds elabsed between `time` and the unix epoch.
+pub fn extract_sec_and_nsec<'tcx>(time: &SystemTime) -> InterpResult<'tcx, (u64, u32)> {
```

Comment:
This could take an `io::Result<SystemTime>`, a `libc_var: &str` and a `mask: &mut ...` to get rid of the rest of the duplication I think

Reply:
mmm yes but shouldn't that be too specific? I can do that but move this to `fs` because that's where it is used.